### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#7]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 ### Initial Assessment

It is understood that the user wants to incorporate wandb run name into the PyTorch Lightning checkpoint filename. As of now, PyTorch Lightning's `ModelCheckpoint` extension only allows for certain metrics and other model related parameters (like epoch number, global step etc.) to be incorporated into the filename of the checkpoint. Wandb's run name is not one of the permitted values.

### Solution Suggestion

We can allow users to provide a name formatting function to the `ModelCheckpoint` which can extract arbitrary properties from their environment. This would let them use not only `wandb.run.name`, but any other custom value they might find useful. 

Here's how the solution may be instantiated:

```python
from pytorch_lightning.callbacks import ModelCheckpoint 
import wandb 

class MyModelCheckpoint(ModelCheckpoint):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def _format_checkpoint_name(self, filename, metrics, ver):
        without_suffix = ".".join(filename.split(".")[:-1])
        return without_suffix.format(wandb_run_name=wandb.run.name, **metrics)

checkpoint_callback = MyModelCheckpoint(filename="{wandb_run_name}-{epoch}-{val_loss:.2f}")
```

In the above snippet, we inherited the `ModelCheckpoint` class and overrode the `_format_checkpoint_name` function to use the wandb run name.

This solution, while accomplishes what the user asks for, goes beyond the currently provided functionalities and might require careful deliberation before integrating into PyTorch Lightning codebase as it increases the complexity of customization. Therefore, it may be best to document this in a tutorial or FAQ for users who might want such a functionality, instead of incorporating it into the codebase. Furthermore, feedback and consensus from other team members should be obtained before proceeding.
